src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVKmerizer.java

```diff
@@ -12,23 +12,23 @@
  * Iterator over successive Kmers from a sequence of characters.
  * Silently skips over parts of the sequence that has characters other than A, C, G, or T.
  */
-public class SVKmerizer implements Iterator<SVKmer> {
+public class SVKmerizer<KType extends SVKmerBase> implements Iterator<KType> {
```

Comment:
Also, do you really need to make this class generic over `KType extends SVKmerBase`? Would it be possible to just make all the return types be `SVKmerBase` instead of `KType`? That way you wouldn't have to do casting in this class. Or would that do something that I'm not seeing like break serialization. Also you wouldn't have to change the subclasses of SVKmerizer.